src/libexpr/parser-state.hh

```diff
@@ -0,0 +1,264 @@
+#pragma once
+
+#include "eval.hh"
+
+namespace nix {
+
+// using C a struct allows us to avoid having to define the special
+// members that using string_view here would implicitly delete.
```

Comment:
Doxygen doc
```suggestion
/**
 * @note Storing a C-style `char *` and `size_t` allows us to avoid
 * having to define the special members that using string_view here
 * would implicitly delete.
 */
```